### PR TITLE
Render email signup locally

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This PR replaces the email signup embed that gets inserted on the page via an iframe with locally rendered components

## Why?
There are several reasons why this is desirable but mainly because this PR removes something like 70Kbs of unused code from the page.

Previously, this embed contained an input and made the api call itself, but now we're only showing some text and a link to the dedicated signup page so there's no need to download the old css, javascript and fonts.

In addition, I've restyled these elements to better match their surroundings, this includes using consistent font sizes and aligning the text better.

## Why not do this?
This removes the ability to change this content dynamically. You can no longer edit the source signup embed content and have that automatically loaded into the iframe. I would argue this pattern is not desirable here and it is better to have these sorts of changes go through a PR review process. It's true that this will add time and effort to making such changes but it also brings benefits.

### Mobile

| Before      | After      |
|-------------|------------|
|mobile-b | mobile-af |

### Tablet

| Before      | After      |
|-------------|------------|
| tablet-b | tab-af |

### Desktop

| Before      | After      |
|-------------|------------|
| desktop-b | Screenshot 2022-05-26 at 09 51 26|

### Wide

| Before      | After      |
|-------------|------------|
| wide-b | w-af|

### CODE REVIEW

**Comment by oliverlloyd** [comment]
> Can it not always just say "Sign up" - looks odd having a long button

Sure, I didn't want to get too editorialisy so tried to keep the original intentions but that amount of text isn't possible in a button in some situations. It'd be great to simplify this though!

@georgeblahblah what do you think?

File: dotcom-rendering/src/web/components/Footer.tsx [hunk 1]
```diff
@@ -263,15 +284,24 @@ export const Footer = ({
...
-				height="100"
-			/>
+			<div css={emailSignup}>
+				<div>
+					All the day's headlines and highlights from the Guardian,
+					direct to you every morning
+				</div>
+				<LinkButton
+					size="small"
+					href="https://www.theguardian.com/info/2015/dec/08/daily-email-uk"
```
[the remark this answers is not in the document]
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/Footer.tsx:294]: Oh interesting, we can absolutely do the same here, so we shall!

[the remark this answers is not in the document]
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/Footer.tsx]: Done 👍

**Comment by oliverlloyd** [comment]
> Can it not always just say "Sign up" - looks odd having a long button

I checked in with the Newsletters team and they asked for "Sign up for our email" so I've updated to show this text everywhere

**Review by mxdvl — APPROVED** [review]
Much cleaner!

File: dotcom-rendering/src/web/components/Footer.tsx [hunk 2]
```diff
@@ -232,6 +236,20 @@ const FooterLinks = ({
...
 const year = new Date().getFullYear();
 
+const decideSignupLink = (edition: Edition): string => {
+	switch (edition) {
+		case 'US':
+			return 'https://www.theguardian.com/info/2015/dec/08/daily-email-us';
+		case 'AU':
+			return 'https://www.theguardian.com/info/2015/dec/08/daily-email-au';
+		case 'INT':
+			return 'https://www.theguardian.com/info/2015/dec/08/daily-email-uk';
```
[the remark this answers is not in the document]
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/Footer.tsx]: It is intentional, yes. There's no international version of the email. I've made a change to make this more obvious though and added a comment

**Review by bryophyta — APPROVED** [review]
This seems great to me! 

Just one question, mainly for my own understanding -- it looks like the css in the iframe source isn't necessary to replicate here, but were any of the class names themselves being used for anything (e.g. as selectors for js scripts)? I just noticed that a couple of them have `js-` prefixes. I can't find any reference to them in our source so I'm guessing it's for some old functionality, but just thought I'd check.

**Comment by oliverlloyd** [comment]
> This seems great to me!
> 
> Just one question, mainly for my own understanding -- it looks like we don't need to replicate the actual css declarations from the iframe source here, but were any of the class names themselves being used for anything (e.g. as selectors for js scripts)? I just noticed that a couple of them have `js-` prefixes. I can't find any reference to them in our source so I'm guessing it's for some old functionality, but just thought I'd check.

Good question!

My assumption here is that all of this css and js was what was previously used when the embed used to be a form. There is an outside possibility that there is an external dependency elsewhere on the page which expects them to be loaded but that would be evil and evil is not allowed

**Comment by bryophyta** [comment]
> > This seems great to me!
> > Just one question, mainly for my own understanding -- it looks like we don't need to replicate the actual css declarations from the iframe source here, but were any of the class names themselves being used for anything (e.g. as selectors for js scripts)? I just noticed that a couple of them have `js-` prefixes. I can't find any reference to them in our source so I'm guessing it's for some old functionality, but just thought I'd check.
> 
> Good question!
> 
> My assumption here is that all of this css and js was what was previously used when the embed used to be a form. There is an outside possibility that there is an external dependency elsewhere on the page which expects them to be loaded but that would be evil and evil is not allowed

sounds good, thanks for clarifying!

**Pull request merged**